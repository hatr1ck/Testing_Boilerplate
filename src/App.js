import React, { Component } from 'react';
import Comp from './components/component'

class App extends Component {
constructor() {
  super()
  this.state = {
   inputText: '',
   test:'state test',
   func(param){
      return param;
    }
  }
  this.foo = (param)=>{
      return param;
    }
    this.onChange = this.onChange.bind(this)
 }



 onChange(e){
  this.setState({
    inputText: e.target.value
  })
 }

  render() {
    return (
      <div>
      <Comp />
      <p id='testProps'>{this.props.testProps}</p>
      <p>{this.state.test}</p>
      <input value='initial' onChange={this.onChange.bind(this)} />
      <button disabled={true}>Disabled button</button>

      </div>
    );
  }
}

export default App;
