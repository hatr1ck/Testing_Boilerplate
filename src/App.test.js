import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, render, mount } from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new Adapter() });

 describe('<App />', ()=>{

		it('test snapsot', () => {
		let component = shallow(<App />);
		expect(component).toMatchSnapshot();
		})

		it('test props', () => {
		let component = shallow(<App testProps='testing prop' />);
		expect(component.find('#testProps').text()).toEqual('testing prop');
		})

		

		it('test state', () => {
		let initialState = {
						inputText:'',
					   test:'state tests',
				 	   func(param){
			 		      return param;
						    }
						  }
		let component = shallow(<App testProps='testing props' />);
		component.setState(initialState)
		expect(component.state()).toEqual(initialState);
		})

		it('test button is disabled', () => {
		let component = shallow(<App />);
		expect(component.find('button').prop('disabled')).toBeTruthy();
		})

		it('test input value', () => {
		let component = shallow(<App />);
		expect(component.find('input').prop('value')).toEqual('initial');
		})

})









 //npm install jest enzyme jest-cli@20.0.4 enzyme-adapter-react-16 axios-mock-adapter --save-dev