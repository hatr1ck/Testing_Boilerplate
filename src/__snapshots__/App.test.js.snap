// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> test snapsot 1`] = `
<div>
  <Comp />
  <p
    id="testProps"
  />
  <p>
    state test
  </p>
  <input
    onChange={[Function]}
    value="initial"
  />
  <button
    disabled={true}
  >
    Disabled button
  </button>
</div>
`;
